tests/integration/test_charm.py

```diff
@@ -26,35 +32,55 @@ async def test_flask_is_up(flask_app, get_unit_ips):
         assert "Hello, World!" in response.text
 
 
+@pytest.mark.parametrize(
+    "update_config, timeout",
+    [
+        pytest.param({"webserver_timeout": 7}, 7, id="timeout=7"),
+        pytest.param({"webserver_timeout": 5}, 5, id="timeout=5"),
+        pytest.param({"webserver_timeout": 3}, 3, id="timeout=3"),
+    ],
+    indirect=["update_config"],
+)
+@pytest.mark.usefixtures("update_config")
+async def test_flask_webserver_timeout(
+    flask_app: Application,
+    get_unit_ips: typing.Callable[[str], typing.Awaitable[tuple[str, ...]]],
+    timeout,
+):
+    """
+    arrange: build and deploy the flask charm, and change the gunicorn timeout configuration.
+    act: send long-running requests to the flask application managed by the flask charm.
+    assert: the gunicorn should restart the worker if the request duration exceeds the timeout.
```

Comment:
For each unit IP the test checks if the request is OK for timeout-1 and then checks if raises an exception for timeout+1. But how is verifying that the worker was restarted? Should it test again with timeout-1?

Reply:
The default timeout is 30 seconds, the change of the timeout value implies that the server has been reloaded.